frontend/src/toolbar/flags/featureFlagsLogic.ts

```diff
@@ -66,9 +66,10 @@ export const featureFlagsLogic = kea<featureFlagsLogicType>([
             (userFlags, localOverrides) => {
                 return userFlags.map((flag) => {
                     const hasVariants = (flag.feature_flag.filters?.multivariate?.variants?.length || 0) > 0
-
                     const currentValue =
-                        flag.feature_flag.key in localOverrides ? localOverrides[flag.feature_flag.key] : flag.value
+                        flag.feature_flag.key in localOverrides
+                            ? localOverrides[flag.feature_flag.key]
+                            : flag.feature_flag.active && flag.value
```

Comment:
why do you need these `.active` checks here and above? Don't seem to be doing anything?

Reply:
Need them here because the "flag.value" that comes back is based on whether the flag matches which seem to return True anyway for flags that are disabled. Which results in a flag having a "on" toggle when it's disabled sooo that's super confusing.

Reply:
If you literally remove it to see, you can see how buggy it is.

Reply:
>  which seem to return True anyway for flags that are disabled.

I think this is the bug to fix then, right?

Ah, it's the same reason as my below comment, it's calling an internal function to get this value which doesn't do the active check, but `get_all_feature_flags` does

Reply:
Well I don't have full context on where `my_flags` is also used in so I didn't want to change it at the API level but if it's just being used in toolbar feature flags then I can make the change there instead of in the frontend.

Reply:
get_all_flags seems to solve the issue fully anyway and I don't really see my_flags being called anywhere else so LGTM?

Reply:
Yep, LGTM! Even if it were being used elsewhere, since you were already removing the `active=True` filter, it would have borked things elsewhere too.

And since it was doing the wrong thing anyway, always better to fix at the source, i.e. endpoint level.